spec/resources/snippet_resource_spec.rb

```diff
@@ -0,0 +1,32 @@
+require 'rails_helper'
+
+RSpec.describe SnippetResource do
+  describe 'Serialization' do
+    it 'serializes attributes' do
+      user = create(:valid_user)
+      snippet = create(:valid_snippet, user: user)
+      resource = SnippetResource.new(user, snippet)
```

Comment:
I'm not sure what these resource classes are for.  Personally I'd find a means to not need them as they complicate things.